pkg/portal/cluster/fetcher.go

```diff
@@ -41,13 +44,28 @@ type client struct {
 // contains Kubernetes clients and returns the frontend-suitable data
 // structures. The concrete implementation of FetchClient wraps this.
 type realFetcher struct {
-	log           *logrus.Entry
-	configCli     configclient.Interface
-	kubernetesCli kubernetes.Interface
-	machineClient machineclient.Interface
+	log              *logrus.Entry
+	configCli        configclient.Interface
+	kubernetesCli    kubernetes.Interface
+	machineClient    machineclient.Interface
+	azureSideFetcher azureSideFetcher
 }
 
-func newRealFetcher(log *logrus.Entry, dialer proxy.Dialer, doc *api.OpenShiftClusterDocument) (*realFetcher, error) {
+type azureSideFetcher struct {
+	resourceGroupName string
+	subscriptionDoc   *api.SubscriptionDocument
+	env               env.Interface
+}
+
+func newAzureSideFetcher(resourceGroupName string, subscriptionDoc *api.SubscriptionDocument, env env.Interface) azureSideFetcher {
+	return azureSideFetcher{
+		resourceGroupName: resourceGroupName,
+		subscriptionDoc:   subscriptionDoc,
+		env:               env,
+	}
+}
+
+func newRealFetcher(log *logrus.Entry, dialer proxy.Dialer, doc *api.OpenShiftClusterDocument, azureazureSideFetcher azureSideFetcher) (*realFetcher, error) {
```

Comment:
Maybe:
```suggestion
func newRealFetcher(log *logrus.Entry, dialer proxy.Dialer, doc *api.OpenShiftClusterDocument, fetcher azureSideFetcher) (*realFetcher, error) {
```

Reply:
So there is another interface called fetcher .. so I don't want someone to confuse that fetcher with this one. This struct is specifically to fetch something from Azure side .. anything in future we want to fetch from Azure, this struct can be used or can be enhanced to fetch more stuff may be.